### PR TITLE
feat(typescript-plugins): use babel to transpile `.ts` plugins

### LINKED ISSUE #51: Add option to transpile TypeScript plugins directly
Probably should use ts-node the same way #31 uses babel-register, but will require some research since the options are obviously different. What should the option be called? `--ts-transpile`? Should we just put both `--babel-transpile` and this one under a single `--transpile-plugins` flag? Maybe it should even be on by default, as long as we’re already shipping the dependencies with the package.

cc @mcMickJuice

### PR DESCRIPTION
**Pull request by eventualbuddha**
Fixes #51 and supersedes #57. The gist of this is that I updated the babel dependencies from 6.x to 7.x, which adds support for transpiling TypeScript. I ended up writing my own `@babel/register`-alike to make it easier to support both `.js` files and `.ts` files.

cc @mcMickJuice @hzoo

### CODE REVIEW

File: src/TransformRunner.ts [hunk 1]
```diff
@@ -47,38 +45,6 @@ export default class TransformRunner {
     return transform(source.content, {
       filename: source.path,
       babelrc: false,
-      parserOpts: {
```
Comment (eventualbuddha) [inline src/TransformRunner.ts]: All this stuff moved into `RecastPlugin` since parser and generator options are now handled by plugin hooks.

**Comment by hzoo** [comment]
> I ended up writing my own @babel/register-alike to make it easier to support both .js files and .ts files.

I think if the functionality is missing we should modify register in a similar way if possible @loganfsmyth

**Comment by eventualbuddha** [comment]
I expect it might be possible to use `@babel/preset-typescript` and `@babel/preset-env` even with `.js` files and have it work. Maybe I’ll just do that.

**Comment by eventualbuddha** [comment]
Actually, the main issue I had was that `ignore` and `only` were getting set even when I didn't provide them. I think the workaround would be for me to specify both of them like so:

```js
  ignore: [/node_modules/],
  only: [/.*/]
```

By default you end up with `ignore` and `only` that restrict it to the current working directory and _not_ `node_modules`, which might be fine except that remote plugins get downloaded to a temp directory which could be anywhere. I'm happy to use `@babel/register` instead if it doesn't require too much futzing to get it working. Using pirates directly is pretty clean, except that I have to hardcode the list of extensions.

**Pull request merged**